lib/builders/graviton-builder.ts

```diff
@@ -0,0 +1,172 @@
+import { BlueprintBuilder } from '../stacks';
+import * as utils from "../utils";
+import * as addons from '../addons';
+import * as spi from '../spi';
+import { MngClusterProvider } from '../cluster-providers';
+import { NestedStack, NestedStackProps } from 'aws-cdk-lib';
+import { Construct } from 'constructs';
+import * as eks from "aws-cdk-lib/aws-eks";
+import * as ec2 from "aws-cdk-lib/aws-ec2";
+
+
+/**
+ * Configuration options for Graviton Builder.
+ */
+export interface GravitonOptions {
+    kubernetesVersion: eks.KubernetesVersion,
+    instanceClass: ec2.InstanceClass,
+    instanceSize: ec2.InstanceSize
+}
+/** 
+ * This builder class helps you prepare a blueprint for setting up 
+ * Graviton nodes with EKS cluster. The `GravitonBuilder` creates the following:
+ * 1. An EKS Cluster` with passed k8s version.
+ * 2. A managed graviton nodegroup for ARM64 software.
+ */
+
+export class GravitonBuilder extends BlueprintBuilder {
+
+
+    public addIstioBaseAddOn(props?: addons.IstioBaseAddOnProps) : GravitonBuilder {
```

Comment:
Hey Ela, what is the point of adding these addons in the builder? Since this extends the BlueprintBuilder customer can just use `builder.addOns(new IstioBaseAddOn())` and similar. Are you trying to isolate the surface area to only the addons validated on Graviton? 

If that is the case, maybe we can come up with a better approach, like using a decorator to add metadata to each addon. 

For example:

```
@arch(ArchType.X86, ArchType.ARM)
export class IstioBaseAddOn ... {
}
```

This decorator will create a global map of all addons that support ARM. In the graviton builder we can query that map and check if any of the registered addons are not supporting ARM and fail fast. 

We can discuss if needed.